lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -105,4 +114,31 @@ public function getLastResponse($force = false)
 
         return ($force) ? $lastResponse : $this->lastCachedResponse;
     }
+
+    /**
+     * Format the path and add query parameters if they exist.
+     *
+     * @param string $path
+     * @param array  $options
+     * @return void
+     */
+    private function formatPath($path, array $options)
+    {
+        $this->path = $path;
+
+        if (array_key_exists('query', $options) && !empty($options['query'])) {
+            $this->path .= '?';
+
+            $i = 0;
+            foreach ($options['query'] as $key => $value) {
```

Comment:
Here is the userland implementation of this code (my code does not handle prefixing numeric keys as I don't need to emulate this case here):

```php
$queryParts = array();
foreach ($options['query'] as $key => $value) {
    $queryParts[] = urlencode($key).'='.urlencodd($value);
}
$query = implode('&', $queryParts);
```

Your own code is missing the urlencoding (which can create the same cache keys for different URLs if my value is ``foo&bar=baz`` and you don't encode it)